gspread/models.py

```diff
@@ -288,6 +288,34 @@ def get_worksheet(self, index):
             return Worksheet(self, properties)
         except (KeyError, IndexError):
             return None
+    
+    def get_worksheet_by_id(self, id):
+        """Returns a worksheet with specified `worksheet id`.
+
+        :param id: The id of a worksheet. it can be seen in the url as the value of the parameter 'gid'.
+        :type id: int
+
+        :returns: an instance of :class:`gspread.models.Worksheet`
+                  or `None` if the worksheet is not found.
+
+        Example. To get first worksheet of a spreadsheet:
+
+        >>> sht = client.open('My fancy spreadsheet')
+        >>> worksheet = sht.get_worksheet_by_id(0):rtype: dict
+        """
+        sheet_data = self.fetch_sheet_metadata()
+
+        try:
+            sheets_length = len(sheet_data['sheets'])
+            print(sheet_data['sheets'])
+            for i in range(0,sheets_length):
+                
+                if id == sheet_data['sheets'][i]['properties']['sheetId']:
+                    properties = sheet_data['sheets'][i]['properties']
+                    return Worksheet(self, properties)
```

Comment:
Here it could be useful to use the same way of iteration as the function [workshet](https://github.com/burnash/gspread/blob/8c31c6df40f51728a82ee26e5b369a765618d191/gspread/models.py#L317) that uses the util function `finditem` and a labmda to filter.

If you need any help with this let me know I can guide you.

Then just below catch the following exceptions `StopIteration` and `KeyError`.

This will make it more consistent accress functions.

Reply:
Used the finditem fuction to perform the operation